packages/http/src/types.ts

```diff
@@ -277,7 +278,7 @@ export interface RouteResolutionOptions extends RouteOptions {
 }
 
 export interface RouteProducerResult {
-  segments: string[];
+  uriTemplate: string;
```

Comment:
```
@route("/bar?fixed=yes{&foo}") op foo(foo: string): void;`
```
what is the uritemplate we will get?

Reply:
Further more, can we have this kind of expression in namespace and interface and add other `@route` definition afterwards ? 
```
@route("/myservice?fixed=yes{&foo}")
namespace MyService {
  @route("/group")
  interface Group {
    @route("/basic")
    @post
    basic(@query foo: string): void; // foo or fixed? 
  }
}
```

Reply:
I have the following questions:
1. is it allowed to have query parameters in the route where the route is not the last segment of this operation? if yes, what does the / mean and how to build the full request url ? 
2. in the above case, do we have one query parameter just foo or two query parameters both fixed and foo ? do we get 
3. in the above case, does it imply an order during serialization between the fixed and foo parameter? however, I have never seen a case where different query parameter order makes difference 
4. can we define extra query parameter if the url template has query parameter included? how to specify that parameter has a url template?

I also notice that we are going to announce this change in August release https://gist.github.com/timotheeguerin/27dd6122c3e01e070f15dd45a64009fc#2-deprecation-deprecate-usage-of-queryformat-in-favor-of-the-new-uri-template-system, I wonder what's the impact on the client emitters?  I guess non of the client emitters are ready for this feature now and it might need to take some time to support, is it okay to announce this feature without client emitters support ?

Reply:
Alll of this is very well defined in the URI template spec. The idea is you can 
```
uriTemplateExpand(template, {
  pathParam1: "abc"
 queryParam1: "def"
}
```
and this just works there is no need to look at anything else.

Now for existing emitter everything will keep wortking for this that worked. The `path` property is returned striped of the uri template annotations and each parameter is described with the correspdonding `explode`, etc.